File: backend/src/main/java/com/apocalypse/thefall/entity/common/BaseNamedEntity.java
package com.apocalypse.thefall.entity.common;


import jakarta.persistence.Column;
import jakarta.persistence.MappedSuperclass;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.util.Map;

@Getter
@Setter
@SuperBuilder
@NoArgsConstructor
@MappedSuperclass
public abstract class BaseNamedEntity extends BaseEntity {

    @Column(name = "code", nullable = false, unique = true)
    private String code;

    @Column(name = "names", columnDefinition = "jsonb", nullable = false)
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, String> names;

    @Column(name = "descriptions", columnDefinition = "jsonb", nullable = false)
    @JdbcTypeCode(SqlTypes.JSON)
    private Map<String, String> descriptions;

    @Column(name = "image_path")
    private String imagePath;

    @Column(name = "display_order", nullable = false)
    @Builder.Default
    private Integer displayOrder = 0;

    @Column(name = "visible", nullable = false)
    @Builder.Default
    private boolean visible = true;
}

File: backend/src/main/java/com/apocalypse/thefall/entity/common/BaseNamedEntity.kt
package com.apocalypse.thefall.entity.common


import jakarta.persistence.Column
import jakarta.persistence.MappedSuperclass
import org.hibernate.annotations.JdbcTypeCode
import org.hibernate.type.SqlTypes

@MappedSuperclass
open class BaseNamedEntity : BaseEntity() {

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "names", columnDefinition = "jsonb", nullable = false)
    open var names: MutableMap<String, String> = mutableMapOf()

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "descriptions", columnDefinition = "jsonb", nullable = false)
    open var descriptions: MutableMap<String, String> = mutableMapOf()

    @Column(name = "image_path")
    open var imagePath: String? = null

    @Column(name = "display_order", nullable = false)
    open var displayOrder: Int = 0

    @Column(name = "visible", nullable = false)
    open var visible: Boolean = true
}

